universal_transfer_operator/transfers/data_providers/filesystem/google/cloud/gcs.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import logging
+from tempfile import NamedTemporaryFile
+
+from airflow.providers.google.cloud.hooks.gcs import GCSHook, _parse_gcs_url
+from data_providers.filesystem.base import BaseFilesystemProviders, TempFile
+from datasets.base import UniversalDataset as Dataset
+
+from constants import FileLocation, LoadExistStrategy, TransferMode
+
+
+class GCSDataProvider(BaseFilesystemProviders):
+    """
+    DataProviders interactions with GS Dataset.
+    """
+
+    def __init__(
+        self,
+        dataset: Dataset,
+        transfer_params: dict = {},
+        transfer_mode: TransferMode = TransferMode.NONNATIVE,
+        if_exists: LoadExistStrategy = "replace",
+    ):
+        super().__init__(
+            dataset=dataset,
+            transfer_params=transfer_params,
+            transfer_mode=transfer_mode,
+            if_exists=if_exists,
+        )
+        self.transfer_mapping: set = {
+            FileLocation.AWS,
+            FileLocation.google_cloud_platform,
+        }
+
+    @property
+    def hook(self) -> GCSHook:
+        """Return an instance of the database-specific Airflow hook."""
+        return GCSHook(
+            gcp_conn_id=self.dataset.conn_id,
+            delegate_to=self.dataset.extra.get("delegate_to", None),
+            impersonation_chain=self.dataset.extra.get("google_impersonation_chain", None),
```

Comment:
I am trying to think if we want to expose `hook_kwargs` -- might be a low-level details.

Not sure yet, just thiking

Reply:
https://github.com/astronomer/astro-sdk/pull/1492/commits/22b0aeeb515b6e9eb72dd68f62a64cb807a5732a